CIP-0035/README.md

```diff
@@ -0,0 +1,302 @@
+| Regressing real performance                                                | SU                 | _Must_ strictly follow a regression in model performance.[^4]                                       |
+
+[^1]: See "Are backwards-compatible binary format changes really safe?".
+[^2]: The binary format change is backwards compatible unless it breaches the limit of how many builtin functions or types can be encoded, in which case that must be changed, forcing a new LV.
+[^3]: See "Why do performance changes require extra steps?".
+[^4]: See "Why do performance changes require extra steps?".
+
+## Specification
+
+### Changes that require a CIP
+
+Changes that require a CIP do not have to each be in an individual CIP, they can be included in batches or in other CIPs. 
+So, for example, a single CIP could propose multiple new builtin functions, or a CIP proposing a change to the ledger could also propose a change to the ledger-script interface.
+
+The requirement to propose a change via a CIP is, as all CIPs are, advisory. 
+In exceptional circumstances or where swift action is required, we expect that changes may still be made without following this process. 
+In such circumstances, a retrospective CIP SHOULD be made after the fact to record the changes and the rationale for them.
+
+#### Additions to the Plutus Core Builtins
+
+Proposals for additions to the set of Plutus Core builtins SHOULD be included in a CIP and should follow both the usual CIP process and the process laid out in this document.
+
+In order to move to Draft status, it MUST include:
+- In the Specification:
+    - Names and types/kinds for the new functions or types.
+    - A source for the implementation (e.g. a library which can be linked against); or a generic description of the functionality which is implementable in any programming language.
+    - In the Rationale:
+    - An argument for the utility of the new builtins.
+
+It SHOULD also include:
+- In the Rationale
+    - Examples of real-world use cases where the new additions would be useful.
+    - A comparison with an implementation using the existing features, and an argument why the builtin is preferable (e.g. better performance).
+
+In order to move to Proposed status, it MUST include:
+- In the Specification:
+    - For new types: a precise description of the measure used for the size of a value of that type.
+    - For new builtin functions: a costing function for the builtin function.
+- In the Rationale:
+    - If an external implementation is provided: an argument that it is trustworthy.
+    - Discussion of how any measures and costing functions were determined.
+- In the Path to Active section:
+    - A named implementor (person or organization) who will implement the support for the new builtins in the Plutus Core interpreter, or a pull request to the `plutus` repository with an implementation.
```

Comment:
Several issues here:

* Again, who decides when the requirements to move to Proposed have been met?
* Based on [here](https://github.com/input-output-hk/plutus/blob/master/plutus-core/cost-model/CostModelGeneration.md) and [here](https://github.com/input-output-hk/plutus/pull/4368#issuecomment-1032025802), the requirement of a costing function being provided is completely unworkable. The only group of people able to cost anything are IOG themselves, and while there is clearly [a goal to extending this](https://github.com/input-output-hk/plutus/issues/4375#issuecomment-1029864939) and [some discussion towards this](https://github.com/input-output-hk/plutus/issues/4384), very non-trivial limitations remain.

Reply:
Implementation may be difficult :) But in general the idea isn't meant to be that the implementor has to do everything and the Plutus team just sit back and chill. As with any contribution, if we think we should accept it we're going to help. In the short term that means that the Plutus team will probably do the costing functions.

(See my main comment for why I didn't include more "implementation process" stuff like this in the text.)

Reply:
Or perhaps a better way of saying it is: there might be named _implementors_ which could be "MLabs and the Plutus team (for cost modelling)"

Reply:
I'd like to see a process for generating cost reports when an implementation is ready to go,  the point should be to _enable_ the community to do this in the correct way, rather than gating it off to specific implementors.   Teach them to fish.  Ideally the request process would not need to go through a CIP editors meeting, but that may become a standard practice.